warehouse/views.py

```diff
@@ -152,29 +153,47 @@ def index(request):
     ],
 )
 def search(request):
-    if request.params.get("q"):
-        query = request.es.query(
-            "multi_match",
-            query=request.params["q"],
-            fields=[
-                "author", "author_email", "description^5", "download_url",
-                "home_page", "keywords^5", "license", "maintainer",
-                "maintainer_email", "normalized_name^10", "platform",
-                "summary^5",
-            ],
-        ).suggest(
-            name="name_suggestion",
-            text=request.params["q"],
-            term={"field": "name"}
-        )
+
+    q = request.params.get("q", '')
+    boosts = {
+        "normalized_name": 10,
+        "description": 5,
+        "keywords": 5,
+        "summary": 5,
+    }
+    fields = [
+        "author", "author_email", "description", "download_url", "home_page",
+        "keywords", "license", "maintainer", "maintainer_email",
+        "normalized_name", "platform", "summary",
+    ]
+
+    should = []
+    if q:
+        for field in fields:
+            kw = {"query": q}
+            if field in boosts:
+                kw["boost"] = boosts[field]
+
+            should.append(Q('match', **{field: kw}))
+
+    # Add a prefix query if ``q`` is longer than one character.
+    if len(q) > 1:
+        should.append(Q('prefix', normalized_name=q))
```

Comment:
Probably want to do something like ``normalized_name=packaging.utils.cannoncalize_name(q)`` here so that we're comparing like to like (normalized to normalized). For that matter, we probably want to do the same on the ``normalized_name`` field up above (Although that's not a regression in this PR, so you don't need to add that unless you want to).